### ISSUE TITLE
Non-English folder and file names are causing file path issues

### ISSUE THREAD

**Issue by PAHXO** [bbilly1/tubearchivist#127]
Hello there, I have an issue with file directory organization, and it is causing problems.
Here are the logs for the crash that happens because of it.

``` rescan subscribed channels
[pid: 46|app: 0|req: 252/252] 172.18.0.2 () {68 vars in 1241 bytes} [Fri Jan  7 21:12:14 2022] POST /process/ => generated 17 bytes in 9 msecs (HTTP/1.1 200) 6 headers in 171 bytes (1 switches on core 0)
[2022-01-07 21:12:14,232: INFO/MainProcess] Task update_subscribed[[*******--feda-41ff-acb0-************] received
[2022-01-07 21:12:14,354: ERROR/ForkPoolWorker-4] Task update_subscribed[*******-feda-41ff-acb0--************] ] raised unexpected: NotADirectoryError(20, 'Not a directory')
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/celery/app/trace.py", line 451, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/celery/app/trace.py", line 734, in __protected_call__
    return self.run(*args, **kwargs)
  File "/app/home/tasks.py", line 59, in update_subscribed
    missing_from_channels = channel_handler.find_missing()
  File "/app/home/src/download.py", line 322, in find_missing
    all_downloaded = pending_handler.get_all_downloaded()
  File "/app/home/src/download.py", line 225, in get_all_downloaded
    videos = os.listdir(channel_path)
NotADirectoryError: [Errno 20] Not a directory: '/youtube/20200331_H8Al********_.mp4' 
```

![image](https://user-images.githubusercontent.com/9639436/148609418-59bb9128-57e0-4614-a359-826ee6246c2d.png)

Also, take a look  at the "-" folder. it hosts multiple channels that use Arabic characters, and it sums them all into one folder while some leak into the /youtube folder which is a root folder.
* note the redacted folders are English and work just fine.

**Comment by bbilly1** [comment]
Yes, I can see how this will become a problem if there are no ascii characters in the channel name. Filename is going to work as far as I can tell, because there is still the timestamp and the video id, even though you won't be able to tell which video that is based on looking just on the filename it will still be able to make a distinct filename per video. 

Originally the thinking here was to strip all none ascii characters except for a few common ones, to get best possible compatibility with different operating systems and file systems.

Honestly, not quite sure how to solve that without breaking peoples installations. But the whole naming functionality will have to be rewritten anyways at some point to make it customizable...

Can you recommend me a all Arabic channel so I can do some testing?

**Comment by UMIQLO** [comment]
Please try this channel, I also got this problem.
https://www.youtube.com/c/TY-BETAch

**Comment by KeithPetro** [comment]
I wonder if it makes sense to use the channel_id for all channel folders instead of channel names. Channel names on YouTube are not unique identifiers, while channel_ids are.

**Comment by bbilly1** [comment]
Originally I had this as /channel-id/video-id.mp4 and be done with it, wouldn't have to deal with any of these complications and only focus on what the interface shows. But this would mean filesystem wise, you wouldn't be able to make anything of it. 

On YouTube as far as I know, you can't have the same channel name for different channels, so in this regard, that's unique. So that shouldn't be a problem. And internally, Tube Archivists only uses the channel id already.

So I think, it's a good compromise now.

**Comment by bbilly1** [comment]
there is a fix for this issue in v0.1.1, delete the affected channels when when you download them again, you'll find the media files in a folder with the channel id.
